File: planning-idsdp-pps-frontend/src/app/main/modules/configuration-management/services/optional-cofog.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {CrudRequestService} from '../../../core/services/crud-request.service';
import {environment} from '../../../../../environments/environment';
import {Observable} from 'rxjs';
import {IOptionalCofog} from '../models/optional-cofog';
import {GET_ACTIVE_OPTIONAL_COFOG, GET_BY_MAIN_COFOG_ID, OPTIONAL_COFOG} from '../constants/optional-cofog.constant';
import {IResponseBody} from '../../../core/models/response';

@Injectable({
    providedIn: 'root'
})
export class OptionalCofogService extends CrudRequestService<IOptionalCofog> {

    baseUrl = '';

    constructor(private http: HttpClient) {
        super(http, OPTIONAL_COFOG);
    }

    // For getting all active Optional Cofog list
    getActiveOptionalCofog(page?: number, size?: number): Observable<IResponseBody<IOptionalCofog>> {
        return this.http.post<IResponseBody<IOptionalCofog>>(this._BASE_URL + GET_ACTIVE_OPTIONAL_COFOG, {page: page, size: size} );
    }

    // For getting all active Optional Cofog list by Main cofog ID
    getByMainCofogId(mainCofogId: number): Observable<IOptionalCofog[]> {
        return this.http.get<IOptionalCofog[]>(this._BASE_URL + GET_BY_MAIN_COFOG_ID + '/' + mainCofogId);
    }

}

File: planning-idsdp-configuration-backend/src/main/java/com/ibcs/idsdp/idsdpconfigration/services/OptionalCofogService.java
package com.ibcs.idsdp.idsdpconfigration.services;

import com.ibcs.idsdp.common.web.dto.request.PageableRequestBodyDTO;
import com.ibcs.idsdp.idsdpconfigration.web.dto.OptionalCofogDTO;
import org.springframework.data.domain.Page;

import java.util.List;

public interface OptionalCofogService {

    Page<OptionalCofogDTO> getActiveOptionalCofog(PageableRequestBodyDTO pageableRequestBodyDTO);

    List<OptionalCofogDTO> getByMainCofogId(long mainCofogId);
}
